tools/scripts/docs_modm_io_generator.py

```diff
@@ -136,56 +138,60 @@ def main():
     with tempfile.TemporaryDirectory(dir=temp_dir) as tempdir:
         tempdir = Path(tempdir)
         modm_path = os.path.abspath(os.path.dirname(sys.argv[0]) + "/../..")
-        print("Modm Path: {}".format(modm_path))
-        print("Temporary directory: {}".format(str(tempdir)))
+        print(f"Modm Path: {modm_path}")
+        print(f"Temporary directory: {tempdir}")
         output_dir = (tempdir / "output")
         (output_dir / "develop/api").mkdir(parents=True)
         os.chdir(tempdir)
         print("Starting to generate documentation...")
         template_overview(output_dir, device_list, board_list, template_path)
-        print("... for {} devices, estimated memory footprint is {} MB".format(len(device_list) + len(board_list), (len(device_list)*70)+2000))
+        print(f"... for {len(device_list) + len(board_list)} devices, estimated memory footprint is {len(device_list)*70+2000} MB")
         with ThreadPool(args.jobs) as pool:
             # We can only pass one argument to pool.map
-            devices = [f"python3 {filepath} --target-job '{modm_path}|{tempdir}|{dev}||{args.deduplicate}'" for dev in device_list]
-            devices += [f"python3 {filepath} --target-job '{modm_path}|{tempdir}|{dev}|{brd}|{args.deduplicate}'" for (brd, dev) in board_list]
-            results = pool.map(lambda d: subprocess.run(d, shell=True).returncode, list(set(devices)))
-        # output_dir.rename(cwd / 'modm-api-docs')
+            devices = [f'python3 {filepath} --target-job "{modm_path}|{tempdir}|{dev}||{args.deduplicate}|{args.compress}"' for dev in device_list]
+            devices += [f'python3 {filepath} --target-job "{modm_path}|{tempdir}|{dev}|{brd}|{args.deduplicate}|{args.compress}"' for (brd, dev) in board_list]
+            devices = list(set(devices))
+            # Run the first generation first so that the other jobs can already deduplicate properly
+            results = [subprocess.call(devices[0], shell=True)]
+            results += pool.map(lambda d: subprocess.call(d, shell=True), devices[1:])
+            # remove all the hash files
+            for file in (output_dir / "develop/api").glob("*.hash"):
+                file.unlink()
         if args.compress:
             print("Zipping docs ...")
-            # Zipping may take more than 10 minutes
-            os.system(f"(cd {str(output_dir)} && {'g' if is_running_on_macos else ''}tar --checkpoint=.100 -czf {str(cwd / 'modm-api-docs.tar.gz')} .)")
-            # shutil.make_archive(str(cwd / 'modm-api-docs'), 'gztar', str(output_dir))
+            # Zipping may take some time
+            tar_cmd = "gtar" if is_running_on_macos else "tar"
+            zip_cmd = f'(cd {str(output_dir)} && {tar_cmd} --checkpoint=.100 -czf {str(cwd)}/modm-api-docs.tar.gz .)'
+            subprocess.call(zip_cmd, shell=True)
```

Comment:
```suggestion
import tarfile

tar = tarfile.open(f"{str(cwd)}/modm-api-docs.tar", "w")
tar.add(str(cwd)}/modm-api-docs.tar.gz)
tar.close()
```

Reply:
Implemented it, will report back how large the `.tar` file is that the CI uploads.

Reply:
1473MB with `.tar` vs 331MB with `.tar.gz`. The upload also does not seem to have been properly deployed on the docs server. So, we're keeping it as is.